### PR TITLE
Added functionality to save link data to a csv file

### PR DESCRIPTION
**Pull request by guillermo136**
This should help us analyze the link extraction logic

### CODE REVIEW

File: src/app/link.service.ts [hunk 1]
```diff
@@ -87,4 +87,24 @@ export class LinkService {
...
+
+    chrome.downloads.download({
+      url: textFile,
+      saveAs: true
+    });
+  }
+
+  private linksToString() {
+    const links = this.linkListSource.getValue();
+    let text = 'url, visible, tag, status, content_type\n';
```
Comment (alifradityar) [inline src/app/link.service.ts:102]: Is it possible to iterate through all metadata fields? so we don't need to manage a separate list here?

File: src/app/expand-collapse-all/expand-collapse-all.component.ts [hunk 2]
```diff
@@ -45,4 +45,8 @@ export class ExpandCollapseAllComponent implements OnInit {
     this.ccdService.updateShowDOMSource(this.showDOMSource);
   }
 
+  exportLinks() {
```
Comment (alifradityar) [inline src/app/expand-collapse-all/expand-collapse-all.component.ts:48]: I'm not sure exportLinks is appropriate in "expand-collapse-all" component. What do you think of changing this into options component or something similar?

File: src/app/link.service.ts [hunk 3]
```diff
@@ -87,4 +87,24 @@ export class LinkService {
     });
   }
 
+  downloadLinks() {
```
Comment (alifradityar) [inline src/app/link.service.ts:90]: I think downloadLinksAsCsvFile is more descriptive

File: src/app/link.service.ts (hunk 1 above)
Comment (guillermo136) [reply src/app/link.service.ts:102]: I don't think we want to save all of the metadata fields though. For example highlightOn and full Source don't seem like things that we want to save.

File: src/app/expand-collapse-all/expand-collapse-all.component.ts (hunk 2 above)
Comment (guillermo136) [reply src/app/expand-collapse-all/expand-collapse-all.component.ts:48]: Taking care of this in a larger refactor branch

**Pull request merged**